Only in R-3.3.2-patched: build
diff -ru R-3.3.2/src/include/Rinternals.h R-3.3.2-patched/src/include/Rinternals.h
--- R-3.3.2/src/include/Rinternals.h	2016-10-04 02:14:57.000000000 -0700
+++ R-3.3.2-patched/src/include/Rinternals.h	2016-12-11 22:48:50.790384685 -0800
@@ -776,7 +776,18 @@
 SEXP Rf_allocList(int);
 SEXP Rf_allocS4Object(void);
 SEXP Rf_allocSExp(SEXPTYPE);
-SEXP Rf_allocVector3(SEXPTYPE, R_xlen_t, R_allocator_t*);
+
+#ifdef __MAIN__
+#define INI_as(v) = v
+#else
+#define INI_as(v)
+#endif
+
+SEXP Rf_allocVector3_impl(SEXPTYPE, R_xlen_t, R_allocator_t*);
+extern SEXP (*Rf_allocVector3)(SEXPTYPE, R_xlen_t, R_allocator_t *) INI_as(Rf_allocVector3_impl);
+
+#define ALLOC_VECTOR_3_IS_A_CALLBACK
+
 R_xlen_t Rf_any_duplicated(SEXP x, Rboolean from_last);
 R_xlen_t Rf_any_duplicated3(SEXP x, SEXP incomp, Rboolean from_last);
 SEXP Rf_applyClosure(SEXP, SEXP, SEXP, SEXP, SEXP);
diff -ru R-3.3.2/src/main/memory.c R-3.3.2-patched/src/main/memory.c
--- R-3.3.2/src/main/memory.c	2016-10-14 15:15:12.000000000 -0700
+++ R-3.3.2-patched/src/main/memory.c	2016-12-11 22:14:54.205985855 -0800
@@ -2456,7 +2456,8 @@
 */
 #define intCHARSXP 73
 
-SEXP allocVector3(SEXPTYPE type, R_xlen_t length, R_allocator_t *allocator)
+
+SEXP Rf_allocVector3_impl(SEXPTYPE type, R_xlen_t length, R_allocator_t *allocator)
 {
     SEXP s;     /* For the generational collector it would be safer to
 		   work in terms of a VECSEXP here, but that would
